### ISSUE TITLE
Suggestion: Cooperate with ROS and make it a open-box would be a good idea

### ISSUE THREAD

**Issue by RidgeSTD** [conatsera/k4a_cuda#2]
Hi! I'm currently working on a project with ROS to remote demonstrate the result of Kinect camera. I found your project interesting but have to manually integrate ROS::PointCloud2 into the CUDA part If you could in your future work separate the CUDA as a library and let the info-source independent( unlike the current one combining CUDA file and K4A driver) would make it a open-box tool, which would be a great time-saver and might come with more starts :)

**Comment by conatsera** [comment]
The data types chosen in this library are deliberate and not with the intent to support any specific application. If you do change them, please ensure you understand the implication of misaligned and oversized copies to the GPU (ROS::PointCloud2 would be both). I recommend alterations to the data structure should occur only after you've received the processed data in your buffer (ie after `GeneratePointCloud()`). With that being said, I'm not entirely sure what your asking here or what open-box means in this context.
